bigquery_etl/cli/query.py

```diff
@@ -1265,6 +1268,150 @@ def render(name, sql_dir, output_dir):
+        ignore_unknown_options=True,
+        allow_extra_args=True,
+    ),
+)
+@click.argument("name")
+@sql_dir_option
+@click.option(
+    "--partition",
+    "-p",
+    help="Partition to check, format <column-name>::<yyy-mm-dd>, must be provided if not executing in Airflow ",
+    type=click.UNPROCESSED,
+    callback=_validate_partition_date,
+    required=False,
+)
+@click.pass_context
+def check(ctx, name, sql_dir, partition):
+    """Run a check."""
+    if not is_authenticated():
+        click.echo(
+            "Authentication to GCP required. Run `gcloud auth login` "
+            "and check that the project is set correctly."
+        )
+        sys.exit(1)
+
+    checks_file, project_id, dataset_id, table = paths_matching_checks_pattern(
+        name, sql_dir, project_id=None
+    )
+
+    _check_query(
+        checks_file,
+        project_id,
+        dataset_id,
+        table,
+        partition,
+        ctx.args,
+    )
+
+
+def _check_query(
+    checks_file,
+    project_id,
+    dataset_id,
+    table,
+    partition,
+    query_arguments,
+):
+    """Run the check."""
+    query_arguments.append("--use_legacy_sql=false")
+    if project_id is not None:
+        query_arguments.append(f"--project_id={project_id}")
+
+    # Partition will be None if from Airflow
+    if partition is None:
+        # Need to check the query_arguments for the value
+        for parameter in query_arguments:
+            if parameter.startswith("--parameter"):
+                param_value = parameter.split("=")[1]
+                partition = _parse_partition_setting(param_value)
+                # once we have a value that passed the date check stop checking.
+                if partition is not None:
+                    break
+    else:
+        # We have a partition value from the CLI so add to the query_arguments.
+        # There should only be 1
+        key, value = next(iter(partition.items()))
+        query_arguments.append(f"--parameter={key}::{value}")
+
+    if partition is None:
+        raise ValueError("No partition specified to check.")
+
+    jinja_params = {
+        **{"project_id": project_id, "dataset_id": dataset_id, "table_name": table},
+        **partition,
+    }
+
+    with tempfile.NamedTemporaryFile(mode="w+") as query_stream:
+        query_stream.write(
+            render_template(
+                checks_file.name,
+                template_folder=str(checks_file.parent),
+                templates_dir="",
+                format=False,
+                **jinja_params,
+            )
+        )
+        query_stream.seek(0)
+
+        # run the query as shell command so that passed parameters can be used as is
+        try:
+            subprocess.check_output(
+                ["bq", "query"] + query_arguments, stdin=query_stream, encoding="UTF-8"
+            )
+        except CalledProcessError as e:
+            print(_parse_check_output(e.output))
+            sys.exit(1)
+
+
```

Comment:
it looks like this moved into `checks.py` and can be removed here?

Reply:
Oops, missed that.